projects/reservoir/ks.py

```diff
@@ -0,0 +1,138 @@
+from dataclasses import dataclass
+from scipy.fftpack import fft, ifft, fftfreq
+from numpy import pi, exp
+import numpy as np
+from typing import Optional
+
+
+def integrate_ks_eqn(
+    Ngrid, domain_size, ic, tmax, timestep,
+):
+    """
+    This code is from
+    kursiv.m - solution of Kuramoto-Sivashinsky equation by ETDRK4 scheme
+    https://github.com/darlliu/follicle/blob/master/mtlab/kursiv.m
+
+    u_t = -u*u_x - u_xx - u_xxxx, periodic BCs on [0,32*pi]
+    computation is based on v = fft(u), so linear term is diagonal
+    compare p27.m in Trefethen, "Spectral Methods in MATLAB", SIAM 2000
+    AK Kassam and LN Trefethen, July 2002
+    """
+    tmax = tmax
+    N = Ngrid
+    d = 1.0 * domain_size / N
+    u = ic
+
+    v = fft(u)
+
+    # Precompute various ETDRK4 scalar quantities:
+    h = timestep  # time step
+
+    nplt = 1
+
+    k = fftfreq(N, d=d)
```

Comment:
Yes- I initially wrote this to assume domain size was normalized by 2 pi when provided to the function, but failed to document it. I'll change this function to have the original domain size units since that is less confusing (ex. I already forgot about the 2 pi factor in the example config in this PR).